OrleansDashboard/Metrics/GrainProfilerFilter.cs

```diff
@@ -78,17 +79,22 @@ private void Track(IIncomingGrainCallContext context, Stopwatch stopwatch, bool
 
         private bool ShouldSkipProfiling(IIncomingGrainCallContext context)
         {
-            try
-            {
-                return
-                    context.Grain.GetType().GetCustomAttribute<NoProfilingAttribute>() != null ||
-                    context.ImplementationMethod?.GetCustomAttribute<NoProfilingAttribute>() != null;
-            }
-            catch (Exception ex)
+            var method = context.ImplementationMethod;
+
+            return shouldProfileCache.GetOrAdd(context.ImplementationMethod, _ =>
             {
-                logger.LogError(100003, ex, "error reading NoProfilingAttribute attribute for grain");
-                return false;
-            }
+                try
+                {
+                    return
+                        context.Grain.GetType().GetCustomAttribute<NoProfilingAttribute>() != null ||
+                        context.ImplementationMethod?.GetCustomAttribute<NoProfilingAttribute>() != null;
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError(100003, ex, "error reading NoProfilingAttribute attribute for grain");
+                    return false;
+                }
```

Comment:
To reduce allocations further, this delegate could be cached. It may require that `context.Grain.GetType().` is replaced with `key.DeclaringType?.`

Reply:
I just have done it to old way without a delegate.

Reply:
Even better